File: ifgi/base/test_OrthonomalBasis.py
#!/usr/bin/env python
#
# Copyright (C) 2010-2011 Yamauchi, Hitoshi
#
#
"""test ifgi orthonomal basis"""

import numpy
import unittest
# From file ILog (first one), import ILog (second one) class
from OrthonomalBasis import OrthonomalBasis

class TestIFGIONB(unittest.TestCase):
    """test: OrthonomalBasis"""

    def test_onb(self):
        """test ONB."""

        onb0 = OrthonomalBasis()
        print onb0
        onb0.init_from_u(numpy.array([1, 0, 0]))
        print onb0

#
# main test
#
if __name__ == '__main__':
    suit0   = unittest.TestLoader().loadTestsFromTestCase(TestIFGIONB)
    alltest = unittest.TestSuite([suit0])
    unittest.TextTestRunner(verbosity=2).run(alltest)

File: ifgi/cpp/base/test_OrthonomalBasis.cpp
//----------------------------------------------------------------------
// ifgi c++ implementation: test_OrthonomalBasis.cpp
// Copyright (C) 2010-2012 Yamauchi, Hitoshi
//----------------------------------------------------------------------
/// \file
/// \brief test_OrthonomalBasis.cpp

#include "OrthonomalBasis.hh"

#include <gtest/gtest.h>

using namespace ifgi;

/// OrthonomalBasis
/// TEST(test_case_name, test_name)
TEST(OrthonomalBasisTest, InitializeTest)
{
    OrthonomalBasis onb0;
    onb0.init_from_u(Scalar_3(1.0, 0.0, 0.0));
    std::cout << onb0.to_string() << std::endl;
}

int main(int argc, char **argv)
{
    ::testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}
